### DISCUSSION TITLE
Include timestamps in navigation tracks

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#124] [Q&A] [answered]
Cruiser - Settings - Routing - Options - Route duration = **SET Speed.**
OR
Cruiser - Settings - Routing - Options - Route duration = **SET Route.**

Condition: 
By **SET Route.**
When the import navigation track contains timestamps these should be respected.

Test.
Calculate a navigation track by the Cruiser app and export than as gpx navigation track.
- Import in Cruiser does not use the attached track/trackpoint timestamps.
- Import in Locus does correctly use the attached track/trackpoint timestamps.
[Cruiser_Brouter.zip](https://github.com/devemux86/cruiser/files/11186219/Cruiser_Brouter.zip)

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the suggestion!

GPX navigation tracks may or may not have time data (or it may not be complete),
so we currently use the selected profile and its average speed to calculate the time.

For example does [RouteYou](https://www.routeyou.com/) provide complete time data in the GPX navigation tracks?

**Comment by devemux86** [comment]
Now I remember that this decision was made a long time ago,
before we introduce the option of duration from route or profile speed.

So now with the option introduced, I will review the implementation. :+1:

**Comment by devemux86** [comment]
It will be available in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
Implemented in [Cruiser 3.0.18](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by 0709wiwiwi** [comment]
Request.
Please add the calculated timestamps also in the simple gpx track export.

**Reply by devemux86** [reply]
Do you mean when export routes?

GPX navigation tracks are complete routes that use a travel mode profile to calculate the time.

GPX normal tracks are just a list of coordinates. What is the use of time here?

Except the recorded GPS tracks that already contain the recorded time.

**Reply by 0709wiwiwi** [reply]
Quite simply this to promote the Cruiser app. Given you easily set your avg speed, Cruiser export tracks with time stamps are useful to have an idea of the ETA. Some (hikers) prefer simple tracks without navigation.  Cruiser does offer simple track export but lacks those very useful reference timestamps.

**Reply by devemux86** [reply]
Cruiser provides time for tracks (either they contain timestamps or not) based on selected app profile and average speed.

Outside Cruiser timestamps on normal GPX tracks can be derived by the third-party apps. Or what use case have in mind?

**Reply by devemux86** [reply]
We can test it in the next Beta.

**Reply by 0709wiwiwi** [reply]
Thanks !
The same simple track with correctly added timestamps is surely just a bit more valuable and convincing. Average speed set: 4.5 km/h.
![Timestamps or not](https://github.com/devemux86/cruiser/assets/54167201/b89d8c06-aa55-42e0-a6b2-31095d814371)

**Reply by devemux86** [reply]
> Please add the calculated timestamps also in the simple gpx track export.

Calculated timestamps in simple GPX track export are available for testing in:

- https://github.com/devemux86/cruiser/discussions/482
